### PR TITLE
Get tests run with subprocess

### PR DESCRIPTION
**Pull request by davidnewpage**
## Description
We would like to manage tests using subprocess so our tests can run in parallel and save our time eventually

## Related Issue
N/A

## Motivation and Context
N/A

## How Has This Been Tested?
Locally. headless must be on

command to run
```
cd tests && python run_test.py
```

### CODE REVIEW

File: tests/plugin_tests/test_js_logs.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        expected_files
+    ), "number of js output files are different"
+    for i in range(len(expected_files)):
+        with open(actual_files[i]) as act:
+            actual_file_content = "".join(act.readlines())
+        with open(expected_files[i]) as exp:
+            expected_file_content = exp.readlines()
+        for item in expected_file_content:
+            assert (
+                item.rstrip("\n") in actual_file_content
```
Comment (swaraj-np) [inline tests/plugin_tests/test_js_logs.py:117]: it should be an exact match between the actual file and the expected. why are you comparing line by line ? it should be a direct equality check of whatever the file read operation returns for the 2 files

File: tests/plugin_tests/test_js_logs.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    test_process = Popen(
+        RUN_NORMAL_TESTS.format(frequency, frequency), shell=True
+    )
+    test_process.wait(TIMEOUT)
+
+    logger.info(f"Start running PLUGIN tests: js_log_frequency={frequency}...")
+    verify_js_logs(frequency)
+
+
+def verify_js_logs(js_log_frequency):
```
Comment (swaraj-np) [inline tests/plugin_tests/test_js_logs.py:49]: this method is not adding any value. the other method already has the same if/else blocks and the current directory can be obtained directly in that method

File: tests/plugin_tests/test_js_logs.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+TIMEOUT = 120  # 120 seconds timeout for running normal tests
+
+RUN_NORMAL_TESTS = "pytest -vv --disable-warnings \
+--headless=True \
+--alluredir='{0}' \
+--report_browser_console_log_capture='{1}' \
+normal_tests"
+
+RUN_PLUGIN_TESTS = "pytest -vv --disable-warnings \
```
Comment (swaraj-np) [inline tests/plugin_tests/test_js_logs.py:28]: why is this needed ?

File: tests/plugin_tests/test_js_logs.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+from shutil import rmtree
+from subprocess import Popen
+import pytest
+import logging
+from tests.step_defs.shared_steps import *  # noqa
+from tests.util import util
+
+logger = logging.getLogger(__name__)
+
+JS_LOG_FREQUENCY = [
```
Comment (swaraj-np) [inline tests/plugin_tests/test_js_logs.py:12]: pls use the actual enum from the code for the plugin instead of duplicating these values here

File: tests/plugin_tests/test_js_logs.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
+import json
+from os import getcwd, path, curdir, makedirs
+from shutil import rmtree
+from subprocess import Popen
+import pytest
+import logging
+from tests.step_defs.shared_steps import *  # noqa
```
Comment (swaraj-np) [inline tests/plugin_tests/test_js_logs.py:7]: why is this needed here ?

File: tests/plugin_tests/test_js_logs.py (hunk 1 above)
Comment (davidnewpage) [reply tests/plugin_tests/test_js_logs.py:117]: @swaraj-np the reason that I'm using contains was because in the actual and expected file, there are some "timestamp" and I couldn't compare them directly. Instead, I use the content of expected file and compare w/ the actual files

File: tests/plugin_tests/test_js_logs.py (hunk 3 above)
Comment (davidnewpage) [reply tests/plugin_tests/test_js_logs.py:28]: what is "this"?
you meant the --report_browser_console_log_capture?

File: tests/plugin_tests/test_js_logs.py (hunk 5 above)
Comment (davidnewpage) [reply tests/plugin_tests/test_js_logs.py:7]: we use Popen to run command in process

File: tests/plugin_tests/test_js_logs.py (hunk 3 above)
Comment (swaraj-np) [reply tests/plugin_tests/test_js_logs.py:21]: what is the confusion here ? the comment is about line 28. there is only a variable initizialization on that line. it is obviously about that variable

File: tests/plugin_tests/test_js_logs.py (hunk 5 above)
Comment (swaraj-np) [reply tests/plugin_tests/test_js_logs.py:7]: you need to import the shared steps module to use `popen` ?

File: tests/plugin_tests/test_js_logs.py (hunk 3 above)
Comment (davidnewpage) [reply tests/plugin_tests/test_js_logs.py:21]: ah my bad... previously I used this to run tests but no need anymore

File: tests/plugin_tests/test_js_logs.py (hunk 1 above)
Comment (swaraj-np) [reply tests/plugin_tests/test_js_logs.py:98]: In that case, I'd prefer removing the timestamp from the strings read from the actual file and doing a direct equals operation with the expected. everything other than the timestamp should match exactly

File: tests/plugin_tests/test_js_logs.py (hunk 5 above)
Comment (davidnewpage) [reply tests/plugin_tests/test_js_logs.py:7]: ah no need - copy paste issue

File: tests/plugin_tests/test_js_logs.py (hunk 1 above)
Comment (davidnewpage) [reply tests/plugin_tests/test_js_logs.py:97]: ok this would take longer time since I don't have strategy for this yet, will try to solve this soon

**Review by swaraj-np — APPROVED** [review]
